src/core/libmaven/mzSample.cpp

```diff
@@ -1256,10 +1256,25 @@ EIC *mzSample::getEIC(float precursorMz, float collisionEnergy, float productMz,
 		}
 		}
 
-		e->scannum.push_back(scan->scannum);
-		e->rt.push_back(scan->rt);
-		e->intensity.push_back(eicIntensity);
-		e->mz.push_back(eicMz);
+		//if rt is already present save the higher intensity for that rt
+		//this can happen when there are multiple product m/z for the same precursor
+		if (!e->rt.empty() && e->rt.back() == scan->rt) {
```

Comment:
Any specific reason to use vector::back?
Here we are assuming that scans are in ascending order, right?

Reply:
That is the assumption as we are going scan by scan. If there is a duplicate value, it will be present at the last index. Using std::find would be wasteful.

Reply:
As pointed out by @sahil21 , this will only work if are assumption , that the scans are sorted, is actually correct.
If it's not sorted, then we will end up inserting duplicates in the EIC

Also, why find() would be wasteful? The complexity will still remain the same. i.e O(n^2)

Reply:
@rish9511 Sorted scans are less an assumption, more of a rule. 
find() searches the whole vector for the value. Back() will just access the last element which is what we need.